### PR TITLE
add auto-update config of Jquery.cycle2

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/malsup/cycle2
**Watch 82
  Star 753
 Fork 215**
@Amomo could you help me check out the PR for #5963 ? thank you~
- [ ] Not found on cdnjs repo
- [x] No already exist issue and PR
- [x] Notable popularity
- [x] Project has public repository on famous online hosting platform (or been hosted on npm) 
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
What's the reason why to delete?

**Comment by maruilian11** [comment]
@PeterDaveHello hmmm... because i modify the folder name to "2.1.5", then it thought "20140415" folder had been deleted.

**Comment by PeterDaveHello** [comment]
This will break the already exist urls, should not do that.

**Comment by maruilian11** [comment]
so, i just need to add `2.1.5`?

**Comment by PeterDaveHello** [comment]
Of course.

**Comment by maruilian11** [comment]
@Amomo  could you help me check this PR? thank you~

**Comment by Amomo** [comment]
Hi @maruilian11 sorry for checking late.

I suggest you to modify the first commit massage to "modify version field to real version number`(2.1.5)` of jquery.cycle2". I think it would be more clearly to read for others.

Could you change it? Thank you~

**Comment by maruilian11** [comment]
@Amomo is that ok?

**Comment by Amomo** [comment]
@maruilian11 you don't pass the CI test~ could you try again?

**Comment by maruilian11** [comment]
@PeterDaveHello please take a look the details of ci.

**Comment by PeterDaveHello** [comment]
You didn't change the version number ...

File: ajax/libs/jquery.cycle2/package.json [hunk 1]
```diff
@@ -19,5 +19,37 @@
...
+    "jquery.cycle2.caption.min.js",
+    "jquery.cycle2.caption2.min.js",
+    "jquery.cycle2.carousel.min.js",
+    "jquery.cycle2.center.min.js",
+    "jquery.cycle2.command.min.js",
+    "jquery.cycle2.core.min.js",
+    "jquery.cycle2.flip.min.js",
+    "jquery.cycle2.hash.min.js",
+    "jquery.cycle2.ie-fade.min.js",
+    "jquery.cycle2.js.map",
```
Comment (PeterDaveHello) [inline ajax/libs/jquery.cycle2/package.json]: We don't generate map file by ourselves ...

File: ajax/libs/jquery.cycle2/package.json [hunk 2]
```diff
@@ -19,5 +19,37 @@
     "type": "git",
     "url": "https://github.com/malsup/cycle2"
   },
-  "license": "MIT"
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/malsup/cycle2.git",
+    "basePath": "src",
```
Comment (PeterDaveHello) [inline ajax/libs/jquery.cycle2/package.json]: There is a `build` folder but you didn't use it, any reason why?

https://github.com/malsup/cycle2/tree/master/build

**Comment by maruilian11** [comment]
@PeterDaveHello do i need to resend a PR?

**Comment by PeterDaveHello** [comment]
@maruilian11 542b10b: fix auto-update config of jquery.cycle2
5da5202: fix structure and missing files in jquery.cycle2,fix #6949

So do you remember what is the reason why?

**Comment by maruilian11** [comment]
@PeterDaveHello sorry, i can't remember, if today, i'll choose `build` not `src`.

**Pull request merged**